src/tools/build_helper/src/git.rs

```diff
@@ -107,6 +109,38 @@ pub fn get_git_merge_base(
     Ok(output_result(git.arg("merge-base").arg(&updated_master).arg("HEAD"))?.trim().to_owned())
 }
 
+/// Searches for the nearest merge commit in the repository.
+///
+/// If it fails to find the upstream remote, it then looks for the most recent commit made
+/// by the merge bot by matching the author's email address with the merge bot's email.
+pub fn get_closest_merge_commit(
+    git_dir: Option<&Path>,
+    config: &GitConfig<'_>,
+    target_paths: &[PathBuf],
+) -> Result<String, String> {
+    let mut git = Command::new("git");
+
+    if let Some(git_dir) = git_dir {
+        git.current_dir(git_dir);
+    }
+
+    let merge_base = get_git_merge_base(config, git_dir).unwrap_or_else(|_| "HEAD".into());
```

Comment:
Oh, so of upstream is available this will start at the upstream merge base and then find the nearest commit from there. The doc comment of this function doesn't mention that so I got confused.

Reply:
With the "If it fails to find the upstream remote," part, it looks quite clear to me actually :thinking: 

https://github.com/rust-lang/rust/blob/5f327176492e3ef66d8140fb771d77bc2b5eebac/src/tools/build_helper/src/git.rs#L115-L119

Reply:
That only explains what it does when it does not find the upstream remote, it says nothing about what it does when it finds the upstream remote.